src/Mod/Arch/archmake/make_reinforcement_linear.py

```diff
@@ -0,0 +1,180 @@
+# ***************************************************************************
+# *   Copyright (c) 2020 Bernd Hahnebach <bernd@bimstatik.org>              *
+# *                                                                         *
+# *   This program is free software; you can redistribute it and/or modify  *
+# *   it under the terms of the GNU Lesser General Public License (LGPL)    *
+# *   as published by the Free Software Foundation; either version 2 of     *
+# *   the License, or (at your option) any later version.                   *
+# *   for detail see the LICENCE text file.                                 *
+# *                                                                         *
+# *   This program is distributed in the hope that it will be useful,       *
+# *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+# *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+# *   GNU Library General Public License for more details.                  *
+# *                                                                         *
+# *   You should have received a copy of the GNU Library General Public     *
+# *   License along with this program; if not, write to the Free Software   *
+# *   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  *
+# *   USA                                                                   *
+# *                                                                         *
+# ***************************************************************************
+
+__title__ = "FreeCAD arch make reinforcement linear"
+__author__ = "Bernd Hahnebach"
+__url__ = "http://www.freecadweb.org"
+
+import FreeCAD
+from FreeCAD import Vector as vec
+
+from DraftTools import translate
+
+# Does it makes sense to put OffsetEnd and OffsetStart in make too?
+# on ifc import they are 0, they will not be needed there
+# they would be needed it direction and distance will be set with and real edge
+# do not put them at the moment, set them to 0, done in init
+
+# would be good to be able to create the object with the make only
+# thus 2 of these 3 are needed, the third is calculated
+# Amount, Distance, Spacing
+#
+# If two are given
+# calculate the third
+#
+# If all are given
+# ATM: print, not yet supported
+# do some test need to be make if there is some rest
+# What if there is a rest.
+# rest:
+# Distance is fixed, fix Amount, change Spacing
+# Amount is fixed, fix Distance, change Spacing
+# Spacing is fixed, fix Amount, change Distance
+# in any case print a message what was changed
+#
+# If only one is given:
+# ATM: print, not yet supported
+#
+# If none of them is given:
+# ATM: print, not yet supported
+#
+
+
+def makeReinforcementLinear(
+    base_rebar,
+    amount=None,
+    spacing=None,
+    distance=None,
+    direction=vec(0, 0, 1),
+    base_placement=FreeCAD.Placement(),
+    name="ReinforcementLinear"
+):
+    """
+    makeReinforcementLinear(
```

Comment:
We should provide the documentation string in proper Numpy style. See `ArchWall.makeWall` for example.

Reply:
this is on ToDo